app/models/document.rb

```diff
@@ -222,6 +224,19 @@ def publish!
     end
   end
 
+  def upload(attachment)
```

Comment:
I have moved this method inside the controller. After some research, many peoples opinion is that private methods inside controllers should not be tested. I feel a bit uncomfortable leaving this method untested?